flexmeasures/data/models/planning/linear_optimization.py

```diff
@@ -339,10 +339,16 @@ def cost_function(m):
     model.costs = Objective(rule=cost_function, sense=minimize)
 
     # Solve
+
+    # load_solutions=False to avoid a RuntimeError exceptions in HiGHS when solving and infeasible problem.
     results = SolverFactory(current_app.config.get("FLEXMEASURES_LP_SOLVER")).solve(
-        model
+        model, load_solutions=False
     )
 
+    # load the results only if the termination conditions is optimal
+    if results.solver.termination_condition == TerminationCondition.optimal:
```

Comment:
I believe there may be other termination conditions where we still want to load the results, such as a timeout with suboptimal solutions having been found already. Maybe the if-condition should check for `!= TerminationCondition.infeasible` instead, or even `len(results) > 0` (to capture the timeout case, which may or may not have found some results already).

Reply:
You're right!